packages/schemer/src/Error.ts

```diff
@@ -30,6 +37,8 @@ export class ValidationError extends ExtendableError {
   }
 }
 
+export type ErrorCode = keyof typeof ErrorCodes;
```

Comment:
IMO an enum named `ErrorCode` would be nicer to use (and works well without TypeScript too since you'd write `ErrorCode.SOME_ERROR`) but I don't think it's a big deal either way.

Reply:
I personally prefer union types, but I think they're both equally nice to use.

When used with TypeScript, autocompletion will be provided for both enums and string literals.

When used with JavaScript, there's also no difference, since neither approach will protect from typos when there's no type checker to detect them.